tests/unit/backend/controllers/public-forms.server.controller.spec.js

```diff
@@ -90,6 +91,21 @@ describe('Public-Forms Controller', () => {
       Controller.redirect(req, res)
     })
 
+    it('should redirect to form with query params retained', (done) => {
+      req.params = {
+        Id: '321564654f65we4f65e4f5',
+      }
+      req.url = '/321564654f65we4f65e4f5?abc=def&zzz=yyy'
+      const uriString = encodeURIComponent('abc=def&zzz=yyy')
+      res.redirect = jasmine.createSpy().and.callFake(() => {
+        expect(res.redirect).toHaveBeenCalledWith(
+          `/#!/321564654f65we4f65e4f5?${uriString}`,
+        )
+        done()
+      })
+      Controller.redirect(req, res)
+    })
```

Comment:
Add more tests for more unicode characters rather than just ASCII. (Tamil, Bengali, Chinese, etc) so we can check if redirect behaviour is retained for those string types.

Reply:
we can add the tests, but i just checked with Li Wei, and the first cut only requires alphanumeric, hyphen, and underscore characters.

Reply:
i've added in the hyphen and underscore to tests just to be sure; will add the other unicode chars in the next cut when we support non-english characters